### DISCUSSION TITLE
overlay issue: Shape.insert_textbox() got an unexpected keyword argument 'overlay'

### DISCUSSION THREAD

**Discussion by andrem-85** [pymupdf/PyMuPDF#3227] [Looking for help] [answered]
Hi all,

I noticed this with insert_text() too. Am I doing something wrong here?

Here is my code:
`shape.insert_textbox(rect, watermark, fontsize=80, color=red, fontname="hebo", morph=(p1, fitz.Matrix(30)), align=1, fill_opacity=.5, overlay=False)`

Throws this:
TypeError: Shape.insert_textbox() got an unexpected keyword argument 'overlay'

Obviously overlay=True doesn't work either.

Not a huge deal but I'd prefer it in the background. Let me know if I'm doing something wrong. Thanks!

**Comment by JorjMcKie** [comment, marked as answer]
The `Page` object has the `Shape` as a child. Accordingly, methods `draw_*()` and `insert_text*()` are wrappers on the page level, which are decomposed in multiple methods on the shape level, e.g. `Page.insert_textbox()` is decomposed into `Shape.insert_textbox()` followed by `Shape.commit()`.
As you can see, only `.commit()` knows the keyword overlay.
The `Page` version **_has_** the additional keyword "overlay".
